ui/js/models/resultsets_store.js

```diff
@@ -910,20 +910,18 @@ treeherder.factory('ThResultSetStore', [
           )
         );
 
-        var getJobGroupInfo = function (job) {
-
-            var name = job.job_group_name;
-            var symbol = job.job_group_symbol;
-            var mapKey = thAggregateIds.getGroupMapKey(job.result_set_id, symbol, tier, job.platform, job.platform_option);
-            var tier;
-
-            if (job.tier && job.tier !== 1) {
-                if (symbol === "?") {
-                    symbol = "";
-                }
-                tier = job.tier;
-            }
-            return { name: name, tier: tier, symbol: symbol, mapKey: mapKey };
+        const getJobGroupInfo = function (job) {
+          const { job_group_name, job_group_symbol, platform, platform_option, result_set_id, tier } = job;
+          let updatedTier;
+          let updatedSymbol = job_group_symbol;
+
+          if (tier && tier !== 1) {
+            updatedSymbol = job_group_symbol === '?' ? '' : job_group_symbol;
+            updatedTier = tier;
```

Comment:
`updatedTier` is always `null` unless tier was 2/3? I'm not sure I follow the reason why this function did it like this?

Reply:
You make a really good point.  Yeah, this is some funky logic.  And the decision on whether or not to display a ``tier`` should be in the front-end code, not here.  So moved it out.